src/Tasks/XmlPeek.cs

```diff
@@ -44,16 +36,8 @@ public class XmlPeek : TaskExtension
         /// <summary>
         /// The XPath Query.
         /// </summary>
-        public string Query
-        {
-            get
-            {
-                ErrorUtilities.VerifyThrowArgumentNull(_query, "Query");
-                return _query;
-            }
-
-            set => _query = value;
-        }
+        [Required]
```

Comment:
Note: _Generally_, adding `[Required]` is a breaking change, but it's ok here because (as @jrdodds pointed out in #9140) failing to specify these lead to immediate unhandled exceptions. The engine-driven "you didn't specify a required parameter" error is much better.